### PR TITLE
Re factor ceph_integration module

### PR DESCRIPTION
**Pull request by shtripat**

### CODE REVIEW

File: tendrl/ceph_integration/flows/create_pool.py [hunk 1]
```diff
@@ -1,6 +1,16 @@
-from tendrl.ceph_integration.flows.flow import Flow
+from tendrl.ceph_integration.manager.crud import Crud
+from tendrl.ceph_integration.manager import utils as manager_utils
+from tendrl.common.flows.base_flow import BaseFlow
 
 
-class CreatePool(Flow):
+class CreatePool(BaseFlow):
     def run(self):
+        self.parameters.update(
```
Comment (cybernth) [inline tendrl/ceph_integration/flows/create_pool.py]: Do we really need to pass this around? Instead directly call these from atoms.

File: tendrl/ceph_integration/flows/delete_pool.py [hunk 2]
```diff
@@ -1,6 +1,16 @@
-from tendrl.ceph_integration.flows.flow import Flow
+from tendrl.ceph_integration.manager.crud import Crud
+from tendrl.ceph_integration.manager import utils as manager_utils
+from tendrl.common.flows.base_flow import BaseFlow
 
 
-class DeletePool(Flow):
+class DeletePool(BaseFlow):
     def run(self):
+        self.parameters.update(
```
Comment (cybernth) [inline tendrl/ceph_integration/flows/delete_pool.py]: Same as above

File: tendrl/ceph_integration/manager/manager.py [hunk 3]
```diff
@@ -114,46 +108,17 @@ def delete_cluster(self, fs_id):
...
-        # self.servers.start()
-
     def join(self):
-        LOG.info("%s joining" % self.__class__.__name__)
-        self._user_request_thread.join()
-        self._discovery_thread.join()
-        self.persister.join()
-        # self.eventer.join()
-        # self.servers.join()
+        super(self).join()
```
Comment (cybernth) [inline tendrl/ceph_integration/manager/manager.py:112]: Should it be super(CephIntegrationManager, self).join() ?

File: tendrl/ceph_integration/manager/manager.py [hunk 4]
```diff
@@ -204,7 +157,7 @@ def main():
                 cluster_id = sys.argv[2]
                 utils.set_tendrl_context(cluster_id)
```
Comment (cybernth) [inline tendrl/ceph_integration/manager/manager.py:158]: Does it create any issues during restart?

File: tendrl/ceph_integration/persistence/persister.py [hunk 5]
```diff
@@ -115,17 +52,85 @@ def update_tendrl_definitions(self, definition):
     def update_pool(self, pool):
         self._store.save(pool)
 
-    def _run(self):
-        LOG.info("Persister listening")
 
-        while not self._complete.is_set():
-            gevent.sleep(0.1)
-            pass
+class CephIntegrationFilePersister(FilePersister):
```
Comment (cybernth) [inline tendrl/ceph_integration/persistence/persister.py]: This can removed as it is not used as of now

**Review by cybernth — CHANGES_REQUESTED** [review]

File: tendrl/ceph_integration/flows/create_pool.py (hunk 1 above)
Comment (shtripat) [reply tendrl/ceph_integration/flows/create_pool.py]: Thats correct. we can import crud module and use the functions as is. Will change the logic accordingly.

File: tendrl/ceph_integration/flows/delete_pool.py (hunk 2 above)
Comment (shtripat) [reply tendrl/ceph_integration/flows/delete_pool.py]: Done

File: tendrl/ceph_integration/manager/manager.py (hunk 3 above)
Comment (shtripat) [reply tendrl/ceph_integration/manager/manager.py:112]: Actually in `__init__()` we use class name and other functions `self` is used

File: tendrl/ceph_integration/manager/manager.py (hunk 4 above)
Comment (shtripat) [reply tendrl/ceph_integration/manager/manager.py:158]: I am not sure at the moment and as we discussed earlier ideally we should be starting bridges as well as services and not child processes of the node agent. We can do this as part of a separate PR preferably as it applies to both gluster and ceph integration modules.

File: tendrl/ceph_integration/persistence/persister.py (hunk 5 above)
Comment (shtripat) [reply tendrl/ceph_integration/persistence/persister.py]: Done. Will remove this.

File: tendrl/ceph_integration/manager/manager.py (hunk 4 above)
Comment (cybernth) [reply tendrl/ceph_integration/manager/manager.py:158]: okay, do we have a git issue to track this?

File: tendrl/ceph_integration/manager/manager.py (hunk 3 above)
Comment (cybernth) [reply tendrl/ceph_integration/manager/manager.py:112]: okay

File: tendrl/ceph_integration/manager/manager.py (hunk 4 above)
Comment (shtripat) [reply tendrl/ceph_integration/manager/manager.py:158]: https://github.com/Tendrl/node_agent/issues/116 introduces starting of integration processes as services and not as command line process.

**Review by r0h4n — CHANGES_REQUESTED** [review]
Fix Travis unit tests, rest LGTM

**Pull request merged**